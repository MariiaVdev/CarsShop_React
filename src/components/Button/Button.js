import React from 'react';
import styles from './Button.module.scss';
import PropTypes from 'prop-types';


class Button extends React.PureComponent {

    render() {
        const { bgColor, textBtn, onClick} = this.props;
        return (
            <>
            <button className={styles.btn} style={{backgroundColor: bgColor}} onClick={onClick}>
                {textBtn}</button>
                    </>
        )
    }
}

Button.propTypes= {
    bgColor: PropTypes.string,
    textBtn: PropTypes.string
}

Button.defaultProps= {
    bgColor: 'blue',
    textBtn: 'Open modal'
}

export default Button;