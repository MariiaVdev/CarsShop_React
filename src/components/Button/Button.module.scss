.btn {
    width: 130px;
    height: 30px;
    float: right;
    margin-right: 0px;
    border: none;
    border-radius: 8px;
    font-size: 14px;
    color: white;
    text-transform: uppercase;
}