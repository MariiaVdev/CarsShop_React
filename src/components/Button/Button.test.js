import {render, screen, fireEvent} from "@testing-library/react"
import Button from './Button'

const handleClick = jest.fn()

describe("Render Button", () => {
    test("should button render with all props", () => {
        const {asFragment} = render(<Button bgColor={'red'} textBtn={'test'}/>);
        expect(asFragment()).toMatchSnapshot()
    })
    test("should button render with default bgColor", () => {
        const {asFragment} = render(<Button textBtn={'test without bg'}/>);
        expect(asFragment()).toMatchSnapshot()
    })
    test("should button render with default text", () => {
        const {asFragment} = render(<Button bgColor={'red'}/>);
        expect(asFragment()).toMatchSnapshot()
    })
    test("should button render with default props", () => {
        const {asFragment} = render(<Button />);
        expect(asFragment()).toMatchSnapshot()
    })
})

describe("Handle click at a button", () => {
    test("should handleClick work", () => {
        render(<Button textBtn={'test'} onClick={handleClick} />);
        const btn = screen.getByText("test");
        fireEvent.click(btn)
        expect(handleClick).toHaveBeenCalled();
    })
})

