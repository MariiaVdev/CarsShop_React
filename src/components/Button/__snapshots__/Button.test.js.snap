// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button should button render with all props 1`] = `
<DocumentFragment>
  <button
    class="btn"
    style="background-color: red;"
  >
    test
  </button>
</DocumentFragment>
`;

exports[`Render Button should button render with default bgColor 1`] = `
<DocumentFragment>
  <button
    class="btn"
    style="background-color: blue;"
  >
    test without bg
  </button>
</DocumentFragment>
`;

exports[`Render Button should button render with default props 1`] = `
<DocumentFragment>
  <button
    class="btn"
    style="background-color: blue;"
  >
    Open modal
  </button>
</DocumentFragment>
`;

exports[`Render Button should button render with default text 1`] = `
<DocumentFragment>
  <button
    class="btn"
    style="background-color: red;"
  >
    Open modal
  </button>
</DocumentFragment>
`;
